web/client-api/src/main/java/io/deephaven/web/client/api/storage/FileContents.java

```diff
@@ -0,0 +1,72 @@
+package io.deephaven.web.client.api.storage;
+
+import elemental2.core.ArrayBuffer;
+import elemental2.core.JsArray;
+import elemental2.dom.Blob;
+import elemental2.promise.Promise;
+import jsinterop.annotations.JsIgnore;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsProperty;
+import jsinterop.annotations.JsType;
+
+import java.util.Objects;
+
+/**
+ * Represents a file's contents loaded from the server. If an etag was specified when loading, client should first test
+ * if the etag of this instance matches - if so, the contents will be empty, and the client's existing contents should
+ * be used.
+ */
+@JsType(namespace = "dh.storage")
+public class FileContents {
+
+    public static FileContents blob(Blob blob) {
+        Objects.requireNonNull(blob, "Blob cannot be null");
+        return new FileContents(blob, null);
+    }
+
+    public static FileContents text(String... text) {
+        return new FileContents(new Blob(JsArray.from(text)), null);
+    }
+
+    public static FileContents arrayBuffers(ArrayBuffer... buffers) {
+        return new FileContents(new Blob(JsArray.from(buffers)), null);
+    }
+
+    private final Blob data;
+
+    private final String etag;
+
+    @JsIgnore
+    public FileContents(Blob data, String etag) {
+        this.data = data;
+        this.etag = etag;
+    }
+
+    @JsIgnore
+    public FileContents(String etag) {
+        this.data = null;
+        this.etag = etag;
+    }
+
+    private Promise<Blob> contents() {
+        if (data != null) {
+            return Promise.resolve(data);
+        }
+        return Promise.reject("No contents available, please use provided etag");
+    }
+
+    @JsMethod
+    public Promise<String> text() {
```

Comment:
Any reason these need to be Promises? We've already got the data...

Reply:
I guess somewhat following the `File` interface?

Reply:
I was aiming at Blob's api (FIle doesnt expose practically anything afaict), but yeah. It also lets us be consistent creating a FileContents and keeping it locally, if you happen to do that.

It might not be ideal if we end up needing to stream results later.